### ISSUE TITLE
Couldn't open the App: libsu NPE regression

### ISSUE THREAD

**Issue by Unknown78** [topjohnwu/Magisk#8147]
Device: Samsung Galaxy Note 4 SM-N910H
Android version: 7.1.1 Nougat (API 25)
Magisk version name: debug
Magisk version code: 27002 ([a89b9e6a](https://github.com/topjohnwu/Magisk/commit/a89b9e6af1dd630d75c977ee0350aeb406738fdd)) [[Download](https://github.com/topjohnwu/Magisk/actions/runs/9576449939)]

**System**: 
- [REFINED_NOTE8_7.1_T2.zip](https://t.me/RefinedNougat/491)
- [Konup_RisingNote4_for_RefinedNote8_7.1.1_ver.2.3.zip](https://xdaforums.com/t/refined-note-8-7-1-1-exynos-updated-mod-risingnote4-ver-2-3.4363583/) 
- [Konup_HomeOneUI_RefinedNote8_7.1.1_ver.2.0_14.07.2022.zip](https://xdaforums.com/t/refined-note-8-7-1-1-exynos-oneui-port-home-screen-bixby-weather-wallpaper-choice-apps.4371353/)
- [Konup_SamsungCamera7_RefinedNote8_7.1.1_v.1.2.zip](https://xdaforums.com/t/refined-note-8-7-1-1-exynos-updated-stock-samsungcamera7-ver-1-2-now-writing-60fps.4201475/)
- [ModemServiceMode_RefinedNote8_7.1.1.zip](https://xdaforums.com/t/refined-note-8-7-1-1-exynos-updated-mod-risingnote4-ver-2-3.4363583/post-88539729)

**Boot**:
- [Helios_Kernel-V3.2-N910C.H-20240107.zip](https://github.com/ananjaser1211/Helios_5433/issues/1#issuecomment-1880082613)

<hr>

I've already direct installed with the debug of [27001 (3d91a561)](https://github.com/topjohnwu/magisk-files/commit/72400d2258acf4a2ec8cc480093ec464fc799ddf)

If I try to update the App to the currently latest debug of [27002 (a89b9e6a)](https://github.com/topjohnwu/Magisk/actions/runs/9576449939)

I'll get this error when trying to launch the App.
```java
06-20 19:57:40.907  3359  4632 E ActivityTrigger: activityStartTrigger: not whiteListedcom.topjohnwu.magisk/com.topjohnwu.magisk.ui.MainActivity/27002
```
[logcat_Magisk_27002_a89b9e6a_app-debug_2024-06-20T19.57.34.924000.txt](https://github.com/user-attachments/files/15914445/logcat_Magisk_27002_a89b9e6a_app-debug_2024-06-20T19.57.34.924000.txt)

I accidently found this issue when troubleshooting other issue, https://github.com/topjohnwu/Magisk/issues/8013?notification_referrer_id=NT_kwDOAX9-N7QxMDQ0MDg2NjE5NzoyNTEzMjU5OQ#issuecomment-2091532113

The problem is now worse than just the App itself gaining root access.
The app itself couldn't launch.

**Comment by yujincheng08** [comment]
regression from cfd1e0cf2266919b91b18c367263885f6fec3322

**Comment by Unknown78** [comment]
How did you know that it was NPE error?

**Comment by aviraxp** [comment]
https://github.com/topjohnwu/libsu/commit/1698dbffefb0e6ccd3bdce8294c7c147d0e2eb14

**Comment by topjohnwu** [comment]
Fixed in 45fa1fce7009a43637d621411b8b95891bbcdfdc

**Comment by Unknown78** [comment]
That's why I asked why this was ruled as NPE error, it seems to yet another different issue.
